hmpps-sqs-spring-boot-autoconfigure/src/test/kotlin/uk/gov/justice/hmpps/sqs/HmppsQueueFactoryTest.kt

```diff
@@ -151,5 +168,21 @@ class HmppsQueueFactoryTest {
       verify(beanFactory).registerSingleton(eq("somequeueid-sqs-dlq-client"), any<SqsAsyncClient>())
       verify(beanFactory).registerSingleton(eq("somequeueid-sqs-listener-factory"), any<HmppsQueueDestinationContainerFactory>())
     }
+
+    @Test
+    fun `should subscribe to topics`() {
+      val someQueueConfig = QueueConfig(queueName = "any", queueAccessKeyId = "any", queueSecretAccessKey = "any", subscribeTopicId = "sometopicid")
+      val hmppsSqsProperties = HmppsSqsProperties(provider = "localstack", queues = mapOf("somequeueid" to someQueueConfig), topics = mapOf("sometopicid" to TopicConfig("arn:aws:sns:1:2:3", "any", "any")))
+      val snsClient = mock<SnsAsyncClient>()
+      val topics = listOf(HmppsTopic("sometopicid", "arn:aws:sns:1:2:3", snsClient))
+
+      hmppsQueues = hmppsQueueFactory.createHmppsQueues(hmppsSqsProperties, topics)
+
+      verify(snsClient).subscribe(
+        check<SubscribeRequest> {
+          assertThat(it.topicArn()).isEqualTo("arn:aws:sns:1:2:3")
+        }
+      )
```

Comment:
And a test to prove that provider=localstack does subscribe to topics - probably covered by the fact that integration tests would fail but it's good to have a counter to to the provider=aws case above